archetype/src/main/resources/META-INF/archetype-post-generate.groovy

```diff
@@ -0,0 +1,142 @@
+import java.nio.file.Files
+import org.apache.commons.io.FileUtils
+
+def jakartaVersion = request.properties["jakartaVersion"].trim()
+def profile = request.properties["profile"].trim()
+def javaVersion = request.properties["javaVersion"].trim()
+def runtime = request.properties["runtime"].trim()
+def docker = request.properties["docker"].trim()
+def outputDirectory = new File(request.getOutputDirectory(), request.getArtifactId())
+
+validateInput(jakartaVersion, javaVersion, runtime, profile, outputDirectory)
+generateRuntime(runtime, jakartaVersion, docker, outputDirectory)
+bindEEPackage(jakartaVersion, outputDirectory)
+generateDocker(docker, runtime, outputDirectory)
+generateMavenWrapper(outputDirectory)
+printSummary()
+
+private validateInput(jakartaVersion, javaVersion, runtime, profile, File outputDirectory){
+    if (profile.equalsIgnoreCase("core") && jakartaVersion != '10') {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, the Core Profile is only supported for Jakarta EE 10")
+    }
+
+    if (runtime.equalsIgnoreCase("payara") && (jakartaVersion != '8') && (javaVersion == '8')) {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, Payara 6 does not support Java SE 8")
+    }    
+
+    if (runtime.equalsIgnoreCase("glassfish") && profile.equalsIgnoreCase("core")) {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, GlassFish does not support the Core Profile")
+    }
+
+    if (runtime.equalsIgnoreCase("glassfish") && (jakartaVersion != '8') && (javaVersion == '8')) {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, GlassFish 7 does not support Java SE 8")
+    }
+
+    if (runtime.equalsIgnoreCase("tomee") && (jakartaVersion == '10')) {
```

Comment:
There are three "if tomee" parts between line 39 and 44. is a sub if not more reasonable here?

Reply:
Do you mean a nested if? If so, yes, easy to do.

Reply:
yes nested :-) makes it easier to understand

Reply:
OK, will address it ASAP.

Reply:
It's addressed. Feel free to mark resolved?